Rover: reject destinations outside fence in GUIDED mode

tridge opened the pull request:
same approach as copter

tridge committed:
Rover: reject destinations outside fence in GUIDED mode

same approach as copter

rmackay9 reviewed (approved):
Looking good!  thanks very much for copying Copter, makes it easy to review

rmackay9 commented:
It looks like it's failing a couple of tests, probably because the test itself needs updating

PolyFence (test fence-related functions) (Did not breach boundary + RTL)
PolyFenceAvoidance

tridge committed:
autotest: fixed rover fence test

we can't use GUIDED mode to demand we drive outside the fence, use
MANUAL instead

peterbarker commented:
tridge did this a second time in https://github.com/ArduPilot/ardupilot/pull/33087 - closing this one.
